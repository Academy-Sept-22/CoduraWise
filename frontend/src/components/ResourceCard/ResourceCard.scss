.resource-card {
  &__container {
    font-family: monospace, sans-serif;
    font-size: 0.9em;
    max-width: 60em;
    margin: 10px 20px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    padding: 5% 0px;
    border-radius: 3px;

    &:hover {
      text-decoration: underline;
      text-decoration-color: white;
      transition: 0.3s;
    }
  }

  &__link {
    background: rgba(171, 41, 41, 0.35);
    box-shadow: 0 8px 32px 0 rgba(31, 38, 135, 0.37);
    backdrop-filter: blur(4px);
    -webkit-backdrop-filter: blur(4px);
    border-radius: 10px;
    border: 1px solid rgba(255, 255, 255, 0.18);

    color: white;
    text-decoration-color: transparent;
    width: 100%;
    padding: 1em;
    display: flex;
    flex-direction: column;
    align-items: center;

    &:hover {
      box-shadow: 0 2px 15px 0 rgb(31 38 135 / 37%);
      transition: 0.2s;
    }
  }
  &__image {
    width: 100%;
    object-fit: cover;
    border-radius: 5px;
    max-height: 9em;
    object-fit: cover;
    filter: grayscale(1);

    &:hover {
      filter: grayscale(0);
      transition: 0.2s;
    }
  }

  &__title {
    margin: 0;
    text-align: center;
  }
}
