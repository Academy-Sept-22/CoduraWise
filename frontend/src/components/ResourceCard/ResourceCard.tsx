import React from "react";
import StarRating from "../StarRating/StarRating";
import "./ResourceCard.scss";

interface ResourceCardProps {
  title: string;
  link: string;
  rating: number;
  mediaType: string;
}

const ResourceCard = ({
  title,
  link,
  rating,
  mediaType,
}: ResourceCardProps) => {
  const imageForCard = (mediaType: string) => {
    if (mediaType === "book") {
      return "book";
    }
    if (mediaType === "article") {
      return "article";
    }
    if (mediaType === "video") {
      return "video";
    }
    if (mediaType === "tutorial") {
      return "tutorial";
    }
    if (mediaType === "course") {
      return "course";
    }
  };

  return (
    <article className="resource-card__container">
      <a
        className="resource-card__link"
        href={link}
        target="_blank"
        rel="noreferrer"
      >
        <img
          className="resource-card__image"
          src={`/ResourceCardImage/${imageForCard(mediaType)}.jpg`}
          alt={imageForCard(mediaType)}
        />
        <StarRating rating={rating} />
        <h3 className="resource-card__title">{title}</h3>
      </a>
    </article>
  );
};

export default ResourceCard;
