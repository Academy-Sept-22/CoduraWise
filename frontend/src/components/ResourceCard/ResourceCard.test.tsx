import TestRenderer from "react-test-renderer";
import ResourceCard from "./ResourceCard";
import React from "react";

test("snapshot test: the html matches", () => {
  const card = TestRenderer.create(<ResourceCard link="url" title="title" />);

  expect(card.toJSON()).toMatchSnapshot();
});
