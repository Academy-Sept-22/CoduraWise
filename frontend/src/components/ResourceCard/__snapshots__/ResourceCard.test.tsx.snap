// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: the html matches 1`] = `
<article>
  <h3>
    Title: 
    title
  </h3>
  <a
    href="url"
  >
    URL: 
    url
  </a>
</article>
`;
